src/uct/base/uct_iface.h

```diff
@@ -396,13 +396,62 @@ typedef struct uct_iface_local_addr_ns {
     }
 
 
+/**
+ * Call TL registration routine.
+ *
+ * @param [in] _name  TL name
+ */
 #define uct_tl_register(_name) \
     uct_tl_##_name##_ctor()
 
 
+/**
+ * Call TL unregistration routine.
+ *
+ * @param [in] _name  TL name
+ */
 #define uct_tl_unregister(_name) \
     uct_tl_##_name##_dtor()
 
+
+/**
+ * Register component and TL
+ *
+ * @param [in] _name  Component and TL name
+ * @param [in] ...    Transport and component registration routines
+ *                    attributes (static)
+ */
+#define UCT_COMP_TL_REGISTER(_name, ...) \
```

Comment:
can we make UCT_COMP_TL_REGISTER and UCT_COMP_TL_UNREGISTER part of UCT_TL_REGISTER_DEF macro?

Reply:
xpmem needs 2 different macros to add [un]initialisation code

Reply:
technically we may use single function for both init/cleanup sequence, just pass init/cleanup parameter into code. then UCT_TL_REGISTER_DEF may have function body for additional sequences